.devcontainer/Dockerfile

```diff
@@ -21,7 +32,7 @@ RUN python3 -m pip install --upgrade pip setuptools && \
 # By default, anything you run in Docker is done as superuser.
 # Conan runs some install commands as superuser, and will prepend `sudo` to
 # these commands, unless `CONAN_SYSREQUIRES_SUDO=0` is in your env variables.
-ENV CONAN_SYSREQUIRES_SUDO 0
+ENV CONAN_SYSREQUIRES_SUDO 1
 # Some packages request that Conan use the system package manager to install
 # a few dependencies. This flag allows Conan to proceed with these installations;
 # leaving this flag undefined can cause some installation failures.
```

Comment:
```suggestion
# See docs at https://docs.conan.io/en/latest/reference/env_vars.html?highlight=conan_sysrequires_mode#conan-sysrequires-sudo
ENV CONAN_SYSREQUIRES_SUDO 1
# Some packages request that Conan use the system package manager to install
# a few dependencies. This flag allows Conan to proceed with these installations;
# leaving this flag undefined can cause some installation failures.
# See docs at https://docs.conan.io/en/latest/reference/env_vars.html?highlight=conan_sysrequires_mode#conan-sysrequires-mode
```

Reply:
I will add this here https://github.com/Jason5480/cpp_boilerplate_project/tree/feature/add_non_root_user_in_docker_container I will close that MR for now as I will not procced any further. I will make a new MR when I find how I can make it work.